assets/js/modules/analytics/datastore/settings.js

```diff
@@ -226,3 +231,40 @@ export function validateCanSubmitChanges( select ) {
 		select( MODULES_ANALYTICS_4 ).__dangerousCanSubmitChanges();
 	}
 }
+
+export const getCanUseSnippet = createRegistrySelector( ( select ) => () => {
+	const analyticsSettings = select( MODULES_ANALYTICS ).getSettings();
+
+	const isTagManagerConnected = select( CORE_MODULES ).isModuleConnected(
+		'tagmanager'
+	);
+
+	if (
+		isTagManagerConnected === undefined ||
+		analyticsSettings === undefined
+	) {
+		return undefined;
+	}
+
+	if ( isTagManagerConnected === null || isTagManagerConnected === false ) {
+		return analyticsSettings.canUseSnippet;
+	}
+
+	const tagManagerUseSnippet = select( MODULES_TAGMANAGER ).getUseSnippet();
+
+	if ( tagManagerUseSnippet === undefined ) {
+		return undefined;
+	}
+
+	if ( tagManagerUseSnippet === false ) {
+		return analyticsSettings.canUseSnippet;
+	}
+
+	const gtmGAPropertyID = select( MODULES_TAGMANAGER ).getGAPropertyID();
+
+	if ( isValidPropertyID( gtmGAPropertyID ) ) {
+		return gtmGAPropertyID !== analyticsSettings.propertyID;
```

Comment:
This was what I had initially. However, @eugene-manuilov suggested that based on AC/IB. As AC wants to return `true` there's a valid `gtmGAPropertyID` but it doesn't matches the `propertyID`. Am I misinterpreting this?

Reply:
@kuasha420 I think my suggestion above is correct. I think the ACs make sense from the perspective of the GA `canUseSnippet` setting value being set on the server and we don't ever call `setCanUseSnippet` otherwise so one might conclude that the setting value could become stale, meaning we should always return a calculated result independent from the saved value because the GTM settings can change without reloading the page.

However, we did add handling for this when saving settings for GTM – if Analytics is connected, it will refetch its settings from the server after saving GTM settings, which will ensure its `canUseSnippet` setting value is accurate based on the current saved GTM configuration.

https://github.com/google/site-kit-wp/blob/c0483f59f121b905f2f971e42b8cd613ad1e3ae5/assets/js/modules/tagmanager/datastore/settings.js#L133-L140

So, from that perspective, I think we only need to calculate when to return `false` here and leave the rest to the saved value. This is very slightly different than this part of the ACs:

```diff
- If they match, it should return `false`, otherwise it should return `true`.
+ If they match, it should return `false`.
```

Does that sound right to you @felixarntz ?

Reply:
@aaemnnosttv I think the part of the ACs you're highlighting (and thus @kuasha420's implementation) is correct as far as I can tell. We need to consider that this is based on which GA property is selected in the Analytics flow, which is purely on the client:
* With your implementation, if the user selects a GA property that matches the GTM one, it still works, but then if they select another one (that doesn't match the GTM one), it would fall back to the value from the server.
* That value on the server could be based on anything, whatever GA property is set on the server. The value may be `false` or `true`. So it may result in the selector returning `false` when the GA property and GTM property do not match, which would not be right.

Reply:
> * That value on the server could be based on anything, whatever GA property is set on the server. The value may be `false` or `true`. So it may result in the selector returning `false` when the GA property and GTM property do not match, which would not be right.

@felixarntz is this not addressed by the GA settings which are updated when we save GTM settings? This happens without requiring a reload so it should be accurate here. The client settings for GTM when editing GA settings can only be the saved settings as any temporary changes are rolled back if not saved when leaving edit context.

Reply:
@aaemnnosttv My point is that the value of `getCanUseSnippet` may also need to change _without_ any settings being saved. For example, when you select another UA property in the dropdown, the toggle should change right away to represent the `canUseSnippet` setting and what it would be set to _if / once_ settings are saved.

Of course the `canUseSnippet` setting is technically static on the client, and the idea here is to replicate the server-side logic on the client, so that we always have a "live" value there.

Reply:
Yeah, I see what you mean. Not sure why I got confused about that, thanks 👍